### PR TITLE
NXCM-3498 NXCM-3499 adding helper Assert class ensuring proper cleanup of Response

### PR DESCRIPTION
**Pull request by peterlynch**
NXCM-3498 NXCM-3499 adding helper assertions ensuring proper cleanup of Response - used on same branch in nexus-enterprise

### CODE REVIEW

File: nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/ResponseAssert.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     * @throws AssertionError if matcher fails
+     */
+    public static void assertThenRelease(final Response actual, final Matcher<Response> matcher) {
+        try
+        {
+            MatcherAssert.assertThat(actual, matcher);
+        }
+        finally
+        {
+            if(actual != null)
```
Comment (velo) [inline nexus/nexus-test-harness/nexus-test-harness-launcher/src/main/java/org/sonatype/nexus/test/utils/ResponseAssert.java]: RequestFacade.releaseResponse() would simplify this code

**Comment by velo** [comment]
+1

**Comment by nabcos** [comment]
+1

**Comment by adreghiciu** [comment]
+1

Are this methods user anywhere? Or is just preparation for some other changes?

**Comment by nabcos** [comment]
@adreghiciu https://github.com/sonatype/nexus-enterprise/pull/25/files#L27R65

**Pull request merged**